leaf-ui/js/displayAnalysis.js

```diff
@@ -198,121 +198,123 @@ function updateNodeValues(nodeID, satValue) {
  * Function to set up the initial analysis configuration upon page load
  */
 function addFirstAnalysisConfig(){
-    // reset to default analysisRequest while preserving userAssignmentsList
-    resetToDefault();
-    // reset highest position to 1
-    highestPosition = 1;
-    // add config 1
-    var id = ("Configuration" + highestPosition);
-    currAnalysisConfig = new AnalysisConfiguration(id, analysisRequest, highestPosition);
-    analysisMap.set(id, currAnalysisConfig);
-    // Add the empty first config to the UI
-    addAnalysisConfig(currAnalysisConfig);
-    refreshAnalysisUI();
+//     // // reset to default analysisRequest while preserving userAssignmentsList
+//     // resetToDefault();
+//     // // reset highest position to 1
+//     // highestPosition = 1;
+//     // // add config 1
+//     // var id = ("Configuration" + highestPosition);
+//     // currAnalysisConfig = new AnalysisConfiguration(id, analysisRequest, highestPosition);
+//     // analysisMap.set(id, currAnalysisConfig);
+//     // // Add the empty first config to the UI
+//     // addAnalysisConfig(currAnalysisConfig);
+//     // refreshAnalysisUI();
+//     configCollection.add(new ConfigModel({name:"Configuration1"}));
+//     configCollection.add(new ConfigModel({name:"Configuration6"}))
 }
 
 /**
  * Function to load analysis configurations and results from JSON into the config sidebar
  * TODO: before running, wipe all other variables and analysis sidebars 
  * so that we can load a new thing after running others
  */
-function loadAnalysis(){
-    // Loop through each configuration
-	for(let config of analysisMap.values()) {
-        // Add the config to the sidebar
-        addAnalysisConfig(config);
-        // Add the results (if any) to the sidebar
-        loadResults(config);
-        // Check if configuration added has higher initialPosition than current highest position, prevents naming duplicates
-        if(config.initialPosition > highestPosition) {
-            highestPosition = config.initialPosition;
-        }
-    }
-    firstConfigElement = document.getElementById('configurations').childNodes[0];
-    currAnalysisConfig = analysisMap.get(firstConfigElement.id);
-    // Set default UAL to preserve in future configs
-    // It is necessary to push each UAL seperately 
-    // to avoid the defaultUAL variable updating along with currAnalysisConfig
-    currAnalysisConfig.userAssignmentsList.forEach(uAL => defaultUAL.push(uAL));
-    analysisRequest = currAnalysisConfig.getAnalysisRequest();
-    switchConfigs(firstConfigElement);
-    // Refresh the sidebar to include the config vars
-    refreshAnalysisUI();
-}
+// function loadAnalysis(){
```

Comment:
It seems like we need to make it so that loadSave functions populates the config/results collections...would that be part of the functionality you want to implement here or later?

Reply:
Yeah, a ton of functionality like load/save is currently broken. This will most likely be one of the last parts we touch, as it will involve reworking some of the conversions to JSON (Since the structure of our data is changing). I think for now the first step is to get all our UI pieces in the right format, and then we can start touching things that talk to the backend.